### PR TITLE
[IOTDB-4399] Control text chunk size in memtable

### PR DESCRIPTION
**Pull request by MrQuansy**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/utils/datastructure/BinaryTVList.java [hunk 1]
```diff
@@ -58,6 +62,12 @@ public void putBinary(long timestamp, Binary value) {
     if (sorted && rowCount > 1 && timestamp < getTime(rowCount - 2)) {
       sorted = false;
     }
+    rawSize += getBinarySize(value);
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/utils/datastructure/BinaryTVList.java:65]: This should already be estimated before write, store the memory in Binary and use here

File: server/src/main/java/org/apache/iotdb/db/utils/datastructure/AlignedTVList.java [hunk 2]
```diff
@@ -582,6 +595,11 @@ protected void releaseLastValueArray() {
     }
   }
 
+  @Override
+  public boolean reachMaxChunkSizeThreshold() {
+    return false;
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/utils/datastructure/AlignedTVList.java:600]: What are you coding? You add a reachMaxChunkSizeFlag but never use it.
Add UTs

**Review by qiaojialin — CHANGES_REQUESTED** [review]

File: server/src/assembly/resources/conf/iotdb-datanode.properties [hunk 3]
```diff
@@ -385,6 +385,11 @@ timestamp_precision=ms
 # Datatype: int
 # avg_series_point_number_threshold=100000
 
+# When a chunk in memtable reaches this threshold, flush the memtable to disk.
+# The default threshold is 20 MB.
+# Datatype: int
```
Comment (HTHou) [inline server/src/assembly/resources/conf/iotdb-datanode.properties:390]: long?

File: server/src/main/java/org/apache/iotdb/db/utils/datastructure/AlignedTVList.java [hunk 4]
```diff
@@ -54,6 +55,8 @@ public abstract class AlignedTVList extends TVList {
   // data types of this aligned tvList
   protected List<TSDataType> dataTypes;
 
+  protected long[] valueChunkRawSize;
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/utils/datastructure/AlignedTVList.java:58]: Add javadoc?

**Review by HTHou — APPROVED** [review]

**Pull request merged**